design/kanister-kopia-integration/kanister-kopia-integration.md

```diff
@@ -0,0 +1,91 @@
+# Integrating Kopia with Kanister
+
+This document proposes all the high-level changes required within Kanister to use [Kopia](https://kopia.io/) as the primary backup and restore tool.
+
+## Motivation
+
+Kanister offers an in-house capability to perform backup and restore to and from object stores using some operation-specific Functions like BackupData, RestoreData, etc.
+Although they are useful and simple to use, these Functions can be significantly improved to provide better reliability, security, and performance.
+
+The improvements would include:
+1. Encryption of data during transfers and at rest
+2. Efficient content-based data deduplication
+3. Configurable data compression
+4. Reduced memory consumption
+5. Increased variety of backend storage target for backups
+
+These improvements can be achieved by using `Kopia` as the primary data movement tool in these Kanister Functions.
+
+Kanister also provides a command line utility `kando` that can be used to move data to and from object stores.
+This tool internally executes `Kopia` commands to move the backup data.
+The v2 version of the example Kanister Blueprints supports this. However, there are a few caveats to using these Blueprints.
+1. `kando` uses `Kopia` only when a Kanister Profile of type `Kopia` is provided
+2. A Kanister Profile of type `Kopia` requires a [Kopia Repository Server](https://kopia.io/docs/repository-server/) running in the same namespace as the Kanister controller
```

Comment:
Do you have an example of what the new server CRD looks like? I am wondering instead of duplicating profile, we can couple/reference the new CRD to a profile?

Reply:
@shlokchaudhari9 could you please post a CRD example when you get a chance?
I sort of agree with @julio-lopez's suggestion of keeping the server CRD independent of the profile CR. This would allow us to slowly deprecate the profiles altogether. There are security benefits as well as some drawbacks around the NFS backend that would be resolved with this approach. But to improve the experience for existing users, the tooling we provide around the creation of the CRD could include options to receive the information from a profile reference.